packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1768,6 +1840,11 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
         assert(summarizeResult.summary.type === SummaryType.Tree,
             0x12f /* "Container Runtime's summarize should always return a tree" */);
 
+        // Reset the decodeHandles. We need to keep track of handles decoded between two summaries.
+        // TODO: The cast to any is done temporarily until the change to IFluidHandleContext in `core-interfaces` is
+        // merged and client is updated to the new `core-interfaces` version.
+        (this.IFluidHandleContext as any).decodedHandles = [];
```

Comment:
This should not be updated here. This should only happen when we get the ack for the summary, similar to how we track other start across summaries.

Reply:
I guess same question as above (on next steps to make your comment not be needed)